### DISCUSSION TITLE
Unable to view HGVS annotations in local install

### DISCUSSION THREAD

**Discussion by lecb** [broadinstitute/seqr#2241] [Q&A] [answered]
**Describe the bug**
I am unable to view HGVS annotations for variants (I've never been able to since installing local seqr version). I've dumped psql references databases and recreated to check that wasn't the reason.

**Link to page(s) where bug is occurring**
All variant pages

**Scope of the bug**
Across all projects

**Screenshots**
![image](https://user-images.githubusercontent.com/23638585/138961170-d8e9d969-c432-440e-9361-24dc37d89d4a.png)

**FYI**
I have upgraded to the latest version of seqr - which has fixed the issues of #2231. As an aside, panelApp isn't working. I note there is a panelapp folder within the docker seqr container. Happy to open a separate issue for this.

**Comment by hanars** [comment]
HGVS annotations come in through the pipeline, so the first thing you need to do is make sure the data is actually in your raw data. This might be something better addressed by me speaking directly with whoever on your end manages data loading

**Comment by eseaby** [comment]
I manage data loading at the moment… we are a small research group doing pretty much everything by ourselves!

**Comment by hanars** [comment]
Good to know! Okay so the first thing I need to know is if you can confirm that the hgvs data is properly added in elasticsearch. In the data in elasticsearch, each variant has a field called `sortedTranscriptConsequences` and for each entry in that list there should be values under `hgvsc` and `hgvsp`

**Comment by lecb** [comment]
Thanks. I'm not sure if I'm looking at the right thing, but using the following command:

`curl 'http://localhost:9200/ibd2021/_search?q=*' | grep 'hgvs'` I can see `NULL` is the result of each hgvsc and hgsvp.

**Comment by hanars** [comment]
You are looking at the right thing, and that means that the issue is actually with the data loading itself, as you are not exporting hgvs annotations. Do you know if your group has made any tweaks on your end to the loading pipeline code? Maybe running vep elsewhere? You could also try making sure you have the latest version of the piepline and then  try running it on a small dataset and see if its now exporting hgvs

**Comment by lecb** [comment]
We have our own pipeline using GATK to generate our joint called vcf. We then put the joint-called vcf directly into seqr using the following command:

```
docker-compose exec pipeline-runner /bin/bash
python3 -m seqr_loading SeqrMTToESTask --local-scheduler \
--reference-ht-path /seqr-reference-data/GRCh38/combined_reference_data_grch38.ht \
--clinvar-ht-path /seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht \
--vep-config-json-path /vep-GRCh38.json \
--es-host elasticsearch \
--es-index-min-num-shards 3 \
--sample-type WES \
--es-index ibd2021 \
--genome-version 38 \
--source-paths /input_vcfs/ibd2021.vcf.gz \
--dest-path /vep_data/ibd2021.mt \
--dont-validate #use this as don't have space to save log files
```

So I'm guessing it must be at this step...

Or are you talking about a different "pipeline"...

**Reply by hanars** [reply]
That is the one I am talking about. Our documentation recommends using a different configuration for vep, so maybe thats where the issue is occurring?
```
 --vep-config-json-path /vep85-GRCh38-loftee-gcloud.json
```

**Reply by lecb** [reply]
A-ha, that must have changed since we first installed... thank you. I'll give that a go.

**Reply by lecb** [reply]
Hmm it didn't work. Looks as if I don't have that file... I get nothing when I run `find / -name vep85-GRCh38-loftee-gcloud.json`

Unsurprisingly, I get an error message when I run the following command `--vep-config-json-path /vep85-GRCh38-loftee-gcloud.json` when loading the pipeline

**ERROR:**
```Traceback (most recent call last):
  File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/hail-elasticsearch-pipelines/luigi_pipeline/seqr_loading.py", line 211, in <module>
    luigi.run() or sys.exit(1)
  File "/usr/local/lib/python3.7/site-packages/luigi/interface.py", line 194, in run
    luigi_run_result = _run(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/luigi/interface.py", line 211, in _run
    return _schedule_and_run([cp.get_task_obj()], worker_scheduler_factory)
  File "/usr/local/lib/python3.7/site-packages/luigi/cmdline_parser.py", line 114, in get_task_obj
    return self._get_task_cls()(**self._get_task_kwargs())
  File "/usr/local/lib/python3.7/site-packages/luigi/task_register.py", line 88, in __call__
    param_values = cls.get_param_values(params, args, kwargs)
  File "/usr/local/lib/python3.7/site-packages/luigi/task.py", line 420, in get_param_values
    raise parameter.MissingParameterException("%s: requires the '%s' parameter to be set" % (exc_desc, param_name))
luigi.parameter.MissingParameterException: SeqrMTToESTask[args=(), kwargs={'reference_ht_path': '/seqr-reference-data/GRCh38/combined_reference_data_grch38.ht', 'clinvar_ht_path': '/seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht', 'vep_config_json_path': '/vep85-GRCh38-loftee-gcloud.json'}]: requires the 'dest_path' parameter to be set
```

I have the latest version of seqr installed as far as I'm aware... 
E

**Reply by hanars** [reply]
That error says you aren't setting the `--dest-path` parameter in your command

**Reply by lecb** [reply]
Sorry, should have said this is the whole command:

```
python3 -m seqr_loading SeqrMTToESTask --local-scheduler \
--reference-ht-path /seqr-reference-data/GRCh38/combined_reference_data_grch38.ht \
--clinvar-ht-path /seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht \
--vep-config-json-path /vep85-GRCh38-loftee-gcloud.json
--es-host elasticsearch \
--es-index-min-num-shards 3 \
--sample-type WES \
--es-index ADupdated \
--genome-version 38 \
--source-paths /input_vcfs/AD_28_joint_called.recalibrated.vcf.gz \
--dest-path /vep_data/ADupdated.mt \
--dont-validate
```

gives the same issue....

**Reply by hanars** [reply]
you're missing a backslash - `--vep-config-json-path /vep85-GRCh38-loftee-gcloud.json` shoudl be `--vep-config-json-path /vep85-GRCh38-loftee-gcloud.json \`

**Reply by lecb** [reply]
Thanks.. unfortunately still throwing a bug..

```  warnings.warn('Parameter "{}" with value "{}" is not of type string.'.format(param_name, param_value))
/usr/local/lib/python3.7/site-packages/luigi/parameter.py:283: UserWarning: Parameter "hgmd_ht_path" with value "None" is not of type string.
  warnings.warn('Parameter "{}" with value "{}" is not of type string.'.format(param_name, param_value))
DEBUG: Checking if SeqrMTToESTask(source_path=<luigi.parameter.Parameter object at 0x7f1942f9d1d0>, use_temp_loading_nodes=True, es_host=elasticsearch, es_port=9200, es_index=adupdated, es_username=pipeline, es_password=None, es_index_min_num_shards=3, source_paths=/input_vcfs/AD_28_joint_called.recalibrated.vcf.gz, dest_path=/vep_data/adupdated.mt, genome_version=38, vep_runner=VEP, reference_ht_path=/seqr-reference-data/GRCh38/combined_reference_data_grch38.ht, clinvar_ht_path=/seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht, hgmd_ht_path=None, sample_type=WES, dont_validate=True, dataset_type=VARIANTS, remap_path=, subset_path=, vep_config_json_path=/vep85-GRCh38-loftee-gcloud.json) is complete
DEBUG: Checking if SeqrVCFToMTTask(source_paths=/input_vcfs/AD_28_joint_called.recalibrated.vcf.gz, dest_path=/vep_data/adupdated.mt, genome_version=38, vep_runner=VEP, reference_ht_path=/seqr-reference-data/GRCh38/combined_reference_data_grch38.ht, clinvar_ht_path=/seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht, hgmd_ht_path=None, sample_type=WES, dont_validate=True, dataset_type=VARIANTS, remap_path=, subset_path=, vep_config_json_path=/vep85-GRCh38-loftee-gcloud.json) is complete
INFO: Informed scheduler that task   SeqrMTToESTask__seqr_reference__VARIANTS__vep_data_adupda_9a4d3de4a5   has status   PENDING
DEBUG: Checking if VcfFile(filename=/input_vcfs/AD_28_joint_called.recalibrated.vcf.gz) is complete
INFO: Informed scheduler that task   SeqrVCFToMTTask__seqr_reference__VARIANTS__vep_data_adupda_c1a1aa2b1a   has status   PENDING
INFO: Informed scheduler that task   VcfFile__input_vcfs_AD_2_025f3d57ac   has status   DONE
INFO: Done scheduling tasks
INFO: Running Worker with 1 processes
DEBUG: Asking scheduler for work...
DEBUG: Pending tasks: 2
INFO: [pid 256] Worker Worker(salt=339970122, workers=1, host=3abbba99a730, username=root, pid=256) running   SeqrVCFToMTTask(source_paths=/input_vcfs/AD_28_joint_called.recalibrated.vcf.gz, dest_path=/vep_data/adupdated.mt, genome_version=38, vep_runner=VEP, reference_ht_path=/seqr-reference-data/GRCh38/combined_reference_data_grch38.ht, clinvar_ht_path=/seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht, hgmd_ht_path=None, sample_type=WES, dont_validate=True, dataset_type=VARIANTS, remap_path=, subset_path=, vep_config_json_path=/vep85-GRCh38-loftee-gcloud.json)
ERROR: [pid 256] Worker Worker(salt=339970122, workers=1, host=3abbba99a730, username=root, pid=256) failed    SeqrVCFToMTTask(source_paths=/input_vcfs/AD_28_joint_called.recalibrated.vcf.gz, dest_path=/vep_data/adupdated.mt, genome_version=38, vep_runner=VEP, reference_ht_path=/seqr-reference-data/GRCh38/combined_reference_data_grch38.ht, clinvar_ht_path=/seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht, hgmd_ht_path=None, sample_type=WES, dont_validate=True, dataset_type=VARIANTS, remap_path=, subset_path=, vep_config_json_path=/vep85-GRCh38-loftee-gcloud.json)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 199, in run
    new_deps = self._run_get_new_deps()
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 141, in _run_get_new_deps
    task_gen = self.task.run()
  File "/hail-elasticsearch-pipelines/luigi_pipeline/seqr_loading.py", line 53, in run
    if self.vep_config_json_path: check_if_path_exists(self.vep_config_json_path, "vep_config_json_path")
  File "/hail-elasticsearch-pipelines/luigi_pipeline/seqr_loading.py", line 18, in check_if_path_exists
    raise ValueError(f"{label} path not found: {path}")
ValueError: vep_config_json_path path not found: /vep85-GRCh38-loftee-gcloud.json
DEBUG: 1 running tasks, waiting for next task to finish
INFO: Informed scheduler that task   SeqrVCFToMTTask__seqr_reference__VARIANTS__vep_data_adupda_c1a1aa2b1a   has status   FAILED
DEBUG: Asking scheduler for work...
DEBUG: Done
DEBUG: There are no more tasks to run at this time
DEBUG: There are 2 pending tasks possibly being run by other workers
DEBUG: There are 2 pending tasks unique to this worker
DEBUG: There are 2 pending tasks last scheduled by this worker
INFO: Worker Worker(salt=339970122, workers=1, host=3abbba99a730, username=root, pid=256) was stopped. Shutting down Keep-Alive thread
INFO: 
===== Luigi Execution Summary =====

Scheduled 3 tasks of which:
* 1 complete ones were encountered:
    - 1 VcfFile(filename=/input_vcfs/AD_28_joint_called.recalibrated.vcf.gz)
* 1 failed:
    - 1 SeqrVCFToMTTask(...)
* 1 were left pending, among these:
    * 1 had failed dependencies:
        - 1 SeqrMTToESTask(...)

This progress looks :( because there were failed tasks```

**Reply by hanars** [reply]
It looks like you don't actually have that VEP config downloaded. Make sure you've downloaded all the prerequisites listed in the "annotate on prem" section of the local deployment readme: https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#option-2-annotate-and-load-on-prem

**Comment by lecb** [comment]
Thanks. This must have changed from when we originally installed seqr?

**Reply by lecb** [reply]
FYI, it looks like the vep data was downloaded from 2019(!) and the loftee data from March 2020. Is the loftee data updated? 

I assume I will have it pay for it as it's using gsutil? We don't have any funding for this so it'll come out of my personal pocket... it'd be useful to know if the loftee data has changed sinced March 2020?
E

**Reply by hanars** [reply]
We make the resources freely available and pay for storage and egress ourselves, so it should not cost you anything to download even if it is using gsutil. If somehow you do find yourself billed, PLEASE let me know and we will reimburse you and fix it, as its supposed to be free

I don't really maintain the VEp/loftee build so I'm not sure if its the most up to date, but it is the one we use on our seqr instance so hopefully its not too far off?

**Reply by lecb** [reply]
Thanks very much. Just to update you, I downloaded and extracted the files again as per the readme...

```
# download VEP reference data
mkdir -p ${SEQR_DIR}/data/vep_data/homo_sapiens
cd ${SEQR_DIR}/data/vep_data
curl -L http://ftp.ensembl.org/pub/release-99/variation/indexed_vep_cache/homo_sapiens_vep_99_GRCh38.tar.gz | tar xzf - &   # for the VEP GRCh38 cache (can have both the GRCh37 and GRCh38 caches at the same time)

mkdir -p ${SEQR_DIR}/data/vep_data/loftee_data/GRCh38/
cd ${SEQR_DIR}/data/vep_data/loftee_data/GRCh38/
gsutil cat gs://seqr-reference-data/vep_data/loftee-beta/GRCh38.tar | tar xf  - & 
```
...and I'm still getting the same issue when trying to load in the data. It seems as if the `vep85-GRCh38-loftee-gcloud.json` is still not found. Any ideas where else this could be or if there is some other vep_data I need to download?

**Reply by hanars** [reply]
can you check and see if maybe it got created in a nested folder relative to where you are doing the data loading?

**Reply by lecb** [reply]
Nope, doesn't exist. My assumption it wouldn't be downloaded from ensembl? It would be somewhere in the seqr-reference-data?

**Reply by hanars** [reply]
potentially. `gsutil cat gs://seqr-reference-data/vep_data/loftee-beta/GRCh38.tar | tar xf  - & ` extracted something to somewhere, so I would also check in there

**Reply by lecb** [reply]
I've checked the entire server... it's nowhere to be seen... 

Just so I understand, is this json file one you generated or one generated by ensembl?

Secondly, ever since I updated to the latest seqr version after issues with not being able to download tables, the red missense and LOF boxes have disappeared from seqr. Is this likely to be solved by re-loading the data with the `vep85-GRCh38-loftee-gcloud.json` file? Or is that likely to be a separate issue? Asking here as I am assuming it is all related... 
E

**Reply by hanars** [reply]
I would assume its all related. @bw2 did the initial set up for getting VEP configured for local installs and wrote all the documentation we've been following, so maybe he has some suggestions of what went wrong and how to get the correct config. I've pretty much exhausted all of my ideas unfortunately

**Reply by hanars** [reply]
Actually, can you change the vep config path back to `/vep-GRCh38.json` now that you've updated the data, and see if that fixes it?

**Reply by lecb** [reply]
The data will load in with `/vep-GRCh38.json` but the hgvs and missense and LOF boxes are still missing... it seems as if all LOEUF and missense constraint data is missing from the gene pages too? 

![image](https://user-images.githubusercontent.com/23638585/140597909-12615048-d38d-4b6f-bede-0397a3781111.png)

**Reply by hanars** [reply]
These are gene specific details - I thought we were talking about variants. Gene reference data comes from a totally different source and is never ingested during the pipeline. For gene level data, run:
```
docker-compose exec seqr python -u manage.py update_all_reference_data --use-cached-omim
```

**Reply by lecb** [reply]
Thanks. The LOF and missense constraint bits are fixed, but the variant level data (hgvs) is not displaying even after re-loading the data... PanelApp doesn't appear either...

**Reply by hanars** [reply]
Hmm so honestly we have kind of hit my limits on my ability to debug issues with the pipeline. Maybe @mike-w-wilson or @bw2 has seen this issue before with VEP not adding hgvs before?

For panel app, thats also a separate command. You need to run the following if you want both PanelApp UK and PanelApp Australia, just run one of them if you only want one 
```
docker-compose exec seqr python -u manage.py import_all_panels https://panelapp.agha.umccr.org/api/v1 --label=AU
docker-compose exec seqr python -u manage.py import_all_panels https://panelapp.genomicsengland.co.uk/api/v1 --label=UK
```

**Reply by lecb** [reply]
Thanks so much Hana. Will wait to see if @bw2 or @mike-w-wilson may be able to help.

**Reply by hanars** [reply]
From another user with a local installation, they found installing some additional hal utilities fixed their problems (https://github.com/broadinstitute/seqr/discussions/2264#discussioncomment-1671879) . Maybe it is worth trying that? You would run 
```
docker-compose exec pipeline python3 -m pip install git+https://github.com/bw2/hail-utils.git
```

**Comment by hanars** [comment, marked as answer]
Hi Ellie,

Someone was able to figure out how to solve the hgvs issue (https://github.com/broadinstitute/seqr/discussions/2695) and the latest release of the docker image should fix the problem. Note that in order to get the fix you will need to re-run the `download_reference_data` step and reload the data

**Reply by lecb** [reply]
This worked! Thanks!
